pkg/kubelet/metrics/metrics.go

```diff
@@ -25,43 +25,68 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-const kubeletSubsystem = "kubelet"
+const (
+	KubeletSubsystem    = "kubelet"
+	PodWorkerKey        = "pod_worker_latency_microseconds"
+	SyncPodsKey         = "sync_pods_latency_microseconds"
+	PodStartKey         = "pod_start_latency_microseconds"
+	PodStatusKey        = "generate_pod_status_latency_microseconds"
+	DockerOperationsKey = "docker_operations_latency_microseconds"
+	DockerErrorsKey     = "docker_errors"
+)
 
 var (
 	ContainersPerPodCount = prometheus.NewSummary(
 		prometheus.SummaryOpts{
-			Subsystem: kubeletSubsystem,
+			Subsystem: KubeletSubsystem,
 			Name:      "containers_per_pod_count",
 			Help:      "The number of containers per pod.",
 		},
 	)
-	SyncPodLatency = prometheus.NewSummaryVec(
+	PodWorkerLatency = prometheus.NewSummaryVec(
 		prometheus.SummaryOpts{
-			Subsystem: kubeletSubsystem,
-			Name:      "sync_pod_latency_microseconds",
+			Subsystem: KubeletSubsystem,
+			Name:      PodWorkerKey,
 			Help:      "Latency in microseconds to sync a single pod. Broken down by operation type: create, update, or sync",
 		},
-		[]string{"operation_type"},
+		[]string{"operation_type", "pod_name"},
```

Comment:
No, today we gather statistics about the operations in general without narrowing down to a specific container.

What you're doing here is probably okay. I get conflicting feedback from the Prometheous folks, but last I understood this will use a bunch of memory (although maybe not noticeable in the short-term) but should still perform ok.